tests/umock_c_reals_int/umock_c_reals_int.c

```diff
@@ -0,0 +1,180 @@
+    {
+        test_on_umock_c_error_calls = new_calls;
+        test_on_umock_c_error_calls[test_on_umock_c_error_call_count].error_code = error_code;
+        test_on_umock_c_error_call_count++;
+    }
+}
+
+#ifdef _MSC_VER
+#pragma warning(disable: 4505)
+#endif
+
+static TEST_MUTEX_HANDLE test_mutex;
+
+BEGIN_TEST_SUITE(umock_c_reals_inttests)
+
+TEST_SUITE_INITIALIZE(suite_init)
+{
+    int result;
+
+    test_mutex = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(test_mutex);
+
+    result = umock_c_init(test_on_umock_c_error);
+    ASSERT_ARE_EQUAL(int, 0, result);
+}
+
+TEST_SUITE_CLEANUP(suite_cleanup)
+{
+    umock_c_deinit();
+
+    TEST_MUTEX_DESTROY(test_mutex);
+}
+
+TEST_FUNCTION_INITIALIZE(test_function_init)
+{
+    int mutex_acquire_result = TEST_MUTEX_ACQUIRE(test_mutex);
+    ASSERT_ARE_EQUAL(int, 0, mutex_acquire_result);
+
+    test_on_umock_c_error_calls = NULL;
+    test_on_umock_c_error_call_count = 0;
+}
+
+TEST_FUNCTION_CLEANUP(test_function_cleanup)
+{
+    umock_c_reset_all_calls();
+
+    free(test_on_umock_c_error_calls);
+    test_on_umock_c_error_calls = NULL;
+    test_on_umock_c_error_call_count = 0;
+
+    TEST_MUTEX_RELEASE(test_mutex);
+}
+
+/* STRICT_EXPECTED_CALL */
+
+TEST_FUNCTION(real_is_called_for_test_dependency_no_args)
+{
+    // arrange
+    int result;
+
+    REGISTER_GLOBAL_MOCK_HOOK(test_dependency_no_args, real_test_dependency_no_args);
+
+    STRICT_EXPECTED_CALL(test_dependency_no_args());
+
+    // act
+    result = test_dependency_no_args();
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, "", umock_c_get_expected_calls());
+    ASSERT_ARE_EQUAL(char_ptr, "", umock_c_get_actual_calls());
+    ASSERT_ARE_EQUAL(int, 42, result);
+}
+
+TEST_FUNCTION(real_is_not_called_for_test_dependency_no_args_no_real)
+{
+    // arrange
+    int result;
+
+    REGISTER_GLOBAL_MOCK_RETURN(test_dependency_no_args_no_real, 1);
+
+    STRICT_EXPECTED_CALL(test_dependency_no_args_no_real());
+
+    // act
+    result = test_dependency_no_args_no_real();
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, "", umock_c_get_expected_calls());
+    ASSERT_ARE_EQUAL(char_ptr, "", umock_c_get_actual_calls());
+    ASSERT_ARE_EQUAL(int, 1, result);
+}
+
+/* Tests_SRS_UMOCK_C_LIB_01_219: [ `REGISTER_GLOBAL_INTERFACE_HOOKS` shall register as mock hooks the real functions for all the functions in a mockable interface. ]*/
+TEST_FUNCTION(reals_are_setup_at_interface_level)
+{
+    // arrange
+    int result;
+
+    REGISTER_GLOBAL_INTERFACE_HOOKS(test_interface);
+
+    STRICT_EXPECTED_CALL(test_dependency_1_arg(45));
+
+    // act
+    result = test_dependency_1_arg(45);
+
+    // assert
+    ASSERT_ARE_EQUAL(char_ptr, "", umock_c_get_expected_calls());
+    ASSERT_ARE_EQUAL(char_ptr, "", umock_c_get_actual_calls());
+    ASSERT_ARE_EQUAL(int, 42, result);
+}
+
+TEST_FUNCTION(real_is_not_called_for_interface_without_reals)
+{
+    // arrange
+    int result;
+
+    REGISTER_GLOBAL_MOCK_RETURN(test_dependency_1_arg_no_real, 1);
```

Comment:
>REGISTER_GLOBAL_MOCK_RETURN(test_dependency_1_arg_no_real, 1); [](start = 4, length = 62)

what's the test behavior in the absence of this line?

Reply:
It would return 0 since the return is a static var.
But I did not want to leave this "undefined" anymore in new tests.

---
In reply to: [293558363](https://github.com/Azure/umock-c/pull/95#discussion_r293558363) [](ancestors = 293558363)